### PR TITLE
FIX - Replace MongoClientOptions to MongoClientUri in mongoClient of Mongo/STHSinks #2387

### PR DESCRIPTION
**Pull request by kumardeepak5**
@AlvaroVega , PR is ready to review.

*Edit*: issue  #2402

Related issue https://github.com/telefonicaid/fiware-cygnus/issues/2387

### CODE REVIEW

**Comment by AlvaroVega** [comment]
This PR sounds goods to me, but not sure we can introduce this breaking change keeping old options (mongo_hosts, mongo_username, mongo_password, mongo_auth_source, mongo_replica_set) at the first time and removing then in a future.

**Comment by fgalan** [comment]
> This PR sounds goods to me, but not sure we can introduce this breaking change keeping old options (mongo_hosts, mongo_username, mongo_password, mongo_auth_source, mongo_replica_set) at the first time and removing then in a future.

I agree. Similar to what we do in Orion, deprecating mongo CLI options in release [3.12.0](https://github.com/telefonicaid/fiware-orion/releases/tag/3.12.0) (but still supported) and removing definitively in a next version [4.0.0](https://github.com/telefonicaid/fiware-orion/releases/tag/4.0.0).

In this sense:

* It's ok to remove that options from documentation (following the rule for deprecated features: _"Documentation on deprecated features is removed from the repository documentation. Documentation is still available in the documentation set associated to older versions"_)
* Recover the support to that options in the code. In the case of using the new `mongo_uri` setting, the old `mongo_` options are ignored.
* Add the following lines to CHANGES_NEXT_RELEASE

```
[cygnus-common][cygnus-ngsi] New setting mongo_uri (#2402)
[cygnus-common][cygnus-ngsi] Deprecate (mongo_hosts, mongo_username, mongo_password, mongo_auth_source, mongo_replica_set) (use mongo_uri instead)
```

**Comment by kumardeepak5** [comment]
@AlvaroVega, @fgalan Please Review PR.

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -1 +1,3 @@
 - [cygnus-ngsi] Upgrade Debian version from 12.5 to 12.6 in Dockerfile
+- [cygnus-common][cygnus-ngsi] New setting mongo_uri (#2402)
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE:2]:
```suggestion
- [cygnus-common][cygnus-ngsi] New setting mongo_uri (#2402, #2387)
```

@AlvaroVega is ok that suggestion?

@kumardeepak5 please wait to @AlvaroVega confirmation before applying the suggestion.

Comment (AlvaroVega) [reply CHANGES_NEXT_RELEASE:2]: #2402 is a PR not a issue
Related issue with this PR is #2387 

CNR entry should be: [cygnus-common][cygnus-ngsi] New setting mongo_uri (#2387)

Comment (fgalan) [reply CHANGES_NEXT_RELEASE]: Fixed in [85e58fb](https://github.com/telefonicaid/fiware-cygnus/pull/2402/commits/85e58fb8eaafe564f4241242748f74d29b9267b3)

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mongo_sink.md [hunk 2]
```diff
@@ -314,6 +314,7 @@ When datamodel changes Cygnus tries to recreate index (delete current and create
 | data\_model | no | dm-by-entity | <i>dm-by-service-path</i>, <i>dm-by-entity</i> or <dm-by-attribute</i>. <i>dm-by-service</i> is not currently supported. |
 | attr\_persistence | no | row | <i>row</i> or <i>column</i>. |
 | attr\_metadata\_store | no | false | <i>true</i> or <i>false</i>. |
+| mongo\_uri | no | <i>empty</i> | Mongo DB Connection String. In case of non empty mongo\_uri parameters (mongo\_hosts, mongo\_username, mongo\_password, mongo\_auth_source, mongo\_replica_set) would be ignored. |
 | mongo\_hosts | no | localhost:27017 | FQDN/IP:port where the MongoDB server runs (standalone case) or comma-separated list of FQDN/IP:port pairs where the MongoDB replica set members run. |
```
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mongo_sink.md:318]:
```suggestion
| mongo\_hosts | no | localhost:27017 | **DEPRECATED** (use mongo_uri instead). FQDN/IP:port where the MongoDB server runs (standalone case) or comma-separated list of FQDN/IP:port pairs where the MongoDB replica set members run. |
```

Please add the same for all the other deprecated fields.

Comment (kumardeepak5) [reply doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_mongo_sink.md]: fixed in commit https://github.com/telefonicaid/fiware-cygnus/pull/2402/commits/016d5ebe4429590759df1ae6be5785a14ed89627

**Review by AlvaroVega — APPROVED** [review]
LGTM

**Review by fgalan — APPROVED** [review]
LGTM. Thanks for your contribution!

**Pull request merged**